py-polars/polars/internals/series/series.py

```diff
@@ -646,6 +646,10 @@ def __getitem__(
         | list[int]
         | list[bool],
     ) -> Any:
+        
+        if not _NUMPY_AVAILABLE:
```

Comment:
Why not to change the condition below which requires numpy to something like `_NUMPY_AVAILABLE and isinstance(item, np.ndarray):` which is already used a few lines below that? Currently a) all further checks for `_NUMPY_AVAILABLE` are no longer necessary after current change, b) the change is much more restrictive than it has to be.